src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1265,81 +1269,200 @@ public void ForceRun()
         #region save/load
 
         /// <summary>
-        /// Opens a Dynamo workspace from a path to an Xml file on disk.
+        /// Opens a Dynamo workspace from a path to a file on disk.
         /// </summary>
         /// <param name="xmlPath">Path to file</param>
         /// <param name="forceManualExecutionMode">Set this to true to discard
         /// execution mode specified in the file and set manual mode</param>
         public void OpenFileFromPath(string xmlPath, bool forceManualExecutionMode = false)
         {
-            var xmlDoc = new XmlDocument();
-            xmlDoc.Load(xmlPath);
+          bool failed = false;
+          try
+          {
+            OpenJsonFileFromPath(xmlPath, forceManualExecutionMode);
```

Comment:
This was discussed in the stand-up meeting yesterday. I am planning on changing all instances of "xmlPath" to "filePath", including where the text is used for logging.